### PR TITLE
INVP-500 Support for v3 elastic data format

### PR DESCRIPTION
**Pull request by mpelzsherman**
Changes needed to accommodate the new format returned by /v3 requests to the onesearch service.

This is needed by rent ruby so we can shut down endeca in production.

Note: the search bar has apparently been broken on the lander pages for quite a while.
Several changes were necessary to get it working, including a change to onesearch.js to handle the new data format.

### CODE REVIEW

File: onesearch.js [hunk 1]
```diff
@@ -282,7 +282,7 @@ define(['jquery'], function(jQuery) {
                             break;
                         case KEY.RETURN:
                             if (selectCurrent()) {
-                                options.onSelect();
+                                //options.onSelect();
```
Comment (Stratus3D) [inline onesearch.js]: Why is this commented out? Is this needed to fix an existing bug?

File: onesearch.js [hunk 2]
```diff
@@ -487,7 +487,8 @@ define(['jquery'], function(jQuery) {
                                 limit: options.max
                             }, extraParams),
                             success: function(data) {
-                                var parsed = options.parse && options.parse(data) || parse(data);
+                                var resultData = data.geo ? data.geo : data;
```
Comment (Stratus3D) [inline onesearch.js]: Do we need this conditional? Is there ever a scenario where V3 data will not have a `geo` key and we will want to return whole response JSON instead?

File: onesearch.js (hunk 1 above)
Comment (mpelzsherman) [reply onesearch.js]: Yes. https://github.com/rentpath/rent/pull/2236

File: onesearch.js [hunk 3]
```diff
@@ -281,12 +281,9 @@ define(['jquery'], function(jQuery) {
...
                         case KEY.RETURN:
-                            if (selectCurrent()) {
-                                options.onSelect();
-                                event.preventDefault();
-                                blockSubmit = true;
-                                return false;
-                            }
+                            event.preventDefault();
+                            blockSubmit = true;
+                            return false;
```
Comment (mpelzsherman) [inline onesearch.js]: Searching using the enter key is broken on rent ruby - the last remaining app to use this lib.

Disabling it here was the quickest solution I could find. We will be porting this code to React soon.

File: onesearch.js (hunk 2 above)
Comment (mpelzsherman) [reply onesearch.js]: I doubt it, but I figured better to keep it backward-compatible.

**Pull request merged**